### ISSUE TITLE
VoiceOver doesn't read MatSelect options inside of a dialog for Safari/Firefox

### ISSUE THREAD

**Issue by jelbourn** [angular/components#20900]
To reproduce: Use the dev-app example for a mat-select inside of a dialog. #20695 fixed this for VoiceOver + Chrome, NVDA, and ChromeVox, but VoiceOver + Safari and Firefox seem to still have issues. 

Related to #20694

**Comment by rameshworsht** [comment]
@jelbourn Has there been consideration to provide the CDK overlay user with the option to attach it to the parent node instead of the body of the page. We could fix this issue if we could attach the select or tooltip on the dialog component rather than to the end of the body at all times.

**Comment by jelbourn** [comment]
That would cause the select panel to be clipped by the dialog.

**Comment by TQuinn33** [comment]
The inconsistent behavior I submitted an issue for is occurring in Chrome, but this ticket is targeted at Safari/Firefox. I am also not sure if this ticket is addressing the same inconsistent behavior described in  #[21787](https://github.com/angular/components/issues/21787), as I don't see any description of the specific behavior being addressed for this ticket.

**Comment by jelbourn** [comment]
@TQuinn33 I believe the issue is the same underlying root cause

**Comment by zarend** [comment]
Hello,

Thank you for participating this this issue. I can no longer reproduce, and I believe this issue has already been fixed.

Environment:
- macOS 13.1 (22C65)
- Firefox version 109.0
- Safari Version 16.2 (18614.3.7.1.5)
- reproduction code: https://github.com/zarend/components/tree/verify-%2320900

-Zach

**Comment by zarend** [comment]
screenshots of above verification ^
<img width="1312" alt="Screenshot 2023-01-23 at 8 54 38 AM" src="https://user-images.githubusercontent.com/7720245/214104210-532b29d3-08f5-444c-aa9d-0bf0bfbbf67b.png">

<img width="1266" alt="Screenshot 2023-01-23 at 8 51 38 AM" src="https://user-images.githubusercontent.com/7720245/214104206-ddc8cd8d-22af-46d9-b372-64aad4d21592.png">
